src/components/SchemaTree.tsx

```diff
@@ -50,9 +42,22 @@ export const SchemaTree = observer<ISchemaTree>(props => {
         striped
         maxRows={maxRows !== undefined ? maxRows + 0.5 : maxRows}
         store={treeStore}
-        rowRenderer={rowRenderer}
+        rowRenderer={(node, rowOptions) => {
```

Comment:
We had to remove the optimization because `rowRendererRight` was not rerendered otherwise.

Reply:
It shouldn't be breaking anything. You just need to specify the dependency for useCallback

Reply:
I agree that it shouldn't but it does. The problem is that: the underlying `rowRendererRight` doesn't re-render (which is, by the way, passed as a React.callback as well). I'm happy to jump on a call to explain what's broken (long story short: checkboxes don't rerender), but I couldn't find any workaround (and passing checkbox's state is not an option).

Reply:
Just leave a todo comment here, I'll revisit it later. Nothing blocking this PR though.

Reply:
We could use `useCallback` having it depend on https://github.com/stoplightio/studio-internal/blob/e65356b86f4162c5ba51ebd0c0b4e219c2ca4e2b/src/components/MaskEditor/index.tsx#L109 but this would mean passing `maskProps` somehow to JSV

Reply:
@P0lip, I've left the TODO. UI is looking fine to me. cc @mallachari